pkg/operation/botanist/vpnseedserver.go

```diff
@@ -0,0 +1,84 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package botanist
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/charts"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/vpnseedserver"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+
+	"k8s.io/utils/pointer"
+)
+
+const diffieHellmanKey = "LS0tLS1CRUdJTiBESCBQQVJBTUVURVJTLS0tLS0KTUlJQkNBS0NBUUVBN2NCWHhHOWFuNktSei9zQjV1aVNPVGY3RWcrdVdWa2hYTzRwZUtEVEFSek1ZYThiN1dSOApCL0F3K0F5VVh0QjN0WHRyemVDNU0zSUhudWhGd01vM0s0b1NPa0ZKeGF0TGxZS2VZMTVyK0t0NXZuT09UM0JXCmVONU9uV2xSNVdpN0daQldiYVFnWFZSNzlONHlzdDQzc1ZoSnVzNkJ5MGxONk9sYzl4RC95czlHSC95a0pWSWgKWi9OTHJ4QUM1bHhqd0NxSk1kOGhycnlDaHVEbHo1OTd2ZzZnWUZ1UlY2MFUvWVU0REs3MUY0SDdtSTA3YUdKOQpsK1NLOFRia0tXRjVJVEk3a1lXYmM0em10ZlhTWGFHak1oTTlvbVFVYVRIOWNzQjk2aHpGSmRlWjRYanh5YlJmClZjM3Q3WFA1cTdhZmVhS21NM0ZoU1hkZUhLQ1RxUXpRdXdJQkFnPT0KLS0tLS1FTkQgREggUEFSQU1FVEVSUy0tLS0tCg=="
+
+// DefaultVPNSeedServer returns a deployer for the vpn-seed-server.
+func (b *Botanist) DefaultVPNSeedServer() (vpnseedserver.VPNSeedServer, error) {
+	imageAPIServerProxy, err := b.ImageVector.FindImage(charts.ImageNameApiserverProxy, imagevector.RuntimeVersion(b.SeedVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
+	}
+
+	imageVPNSeedServer, err := b.ImageVector.FindImage(charts.ImageNameVpnSeedServer, imagevector.RuntimeVersion(b.SeedVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
+	}
+
+	var kubeAPIServerHost *string
+	if b.APIServerSNIEnabled() {
+		kubeAPIServerHost = pointer.StringPtr(b.outOfClusterAPIServerFQDN())
+	}
+
+	return vpnseedserver.New(
+		b.K8sSeedClient.Client(),
+		b.Shoot.SeedNamespace,
+		imageAPIServerProxy.String(),
+		imageVPNSeedServer.String(),
+		kubeAPIServerHost,
+		b.Shoot.Networks.Services.String(),
+		b.Shoot.Networks.Pods.String(),
+		b.Shoot.GetNodeNetwork(),
+		b.Shoot.GetReplicas(1),
+	), nil
+}
+
+func (b *Botanist) DeployVPNServer(ctx context.Context) error {
+
+	if b.Shoot.KonnectivityTunnelEnabled || !b.Shoot.ReversedVPNEnabled {
+		return b.Shoot.Components.ControlPlane.VPNSeedServer.Destroy(ctx)
+	}
+
+	dhKey, err := utils.DecodeBase64(diffieHellmanKey)
+	if err != nil {
+		return err
+	}
+	openvpnDiffieHellmanSecret := map[string][]byte{"dh2048.pem": dhKey}
+	if dh, ok := b.Secrets[v1beta1constants.GardenRoleOpenVPNDiffieHellman]; ok {
+		openvpnDiffieHellmanSecret = dh.Data
+	}
+
+	b.Shoot.Components.ControlPlane.VPNSeedServer.SetSecrets(vpnseedserver.Secrets{
+		TLSAuth: component.Secret{Name: "vpn-seed-server-tlsauth", Checksum: b.CheckSums["vpn-seed-server-tlsauth"], Data: b.Secrets["vpn-seed-server-tlsauth"].Data},
```

Comment:
Can you define a constant for `vpn-seed-server-tlsauth` and use it at the various places?